contributing.md

```diff
@@ -13,10 +13,14 @@ machine to develop Bazel and, when you've made a patch, how to submit it.</p>
 In general, we prefer contributions that fix bugs or add features (as opposed to
 stylistic, refactoring, or "cleanup" changes).
 
-Please check with us on the [dev
-list](https://groups.google.com/forum/#!forum/bazel-dev) before investing a lot
-of time in a patch. Meet Bazel maintainers and other contributors on
-[Slack](https://slack.bazel.build).
+Please check with us on the [dev list](
+https://groups.google.com/forum/#!forum/bazel-dev) before investing
+a lot of time in a patch. You might also meet Bazel maintainers and
```

Comment:
In general, a GitHub issue is the best place to discuss the issue, how to solve it, and if it's worth spending time on it. Asking bazel-dev is useful only in a few rare cases.

Reply:
You are right.  As I looked, I realize the entire section gave unclear or conflicting advice. I reworked it substantially. I also added a section at the end about the various communications channels. Slack is now there, because it should not be in the middle of the standard patch review process. Slack is ephemeral, and we want discussion of how we made design choices in a permanent record.